content/blog/2021-02-18-dvc-2-0-pre-release.md

```diff
@@ -0,0 +1,562 @@
+
+```yaml
+vars:
+    train_matrix: train.pkl
+    test_matrix: test.pkl
+    seed: 20210215
+
+...
+
+stages:
+    process:
+        cmd: python process.py \
+                --seed ${seed} \
+                --train ${train_matrix} \
+                --test ${test_matrix}
+        outs:
+        - ${test_matrix}
+        - ${train_matrix}
+
+        ...
+
+    train:
+        cmd: python train.py ${train_matrix} --seed ${seed}
+        deps:
+        - ${train_matrix}
+```
+
+Also, it gives an ability to localize all important parameters in a single
+`vars` block, and play with them. This is a natural thing to do for scenarios
+like NLP or when hyperparameter optimization is happening not only in the model
+training code but in the data processing as well.
+
+### Pipeline parameters from params files
+
+It is quite common to define pipeline parameters in a config file or a
+parameters file (like `params.yaml`) instead of in the pipeline file `dvc.yaml`
+itself. These parameters defined in `params.yaml` can also be used in
+`dvc.yaml`.
+
+```yaml
+# params.yaml
+models:
+  us:
+    thresh: 10
+    filename: 'model-us.hdf5'
+```
+
+```yaml
+# dvc.yaml
+stages:
+  build-us:
+    cmd: >-
+      python script.py
+        --out ${models.us.filename}
+        --thresh ${models.us.thresh}
+    outs:
+      - ${models.us.filename}
+```
+
+DVC properly tracks params dependencies for each stage starting from the
+previous DVC version 1.0. See the
+[`--params` option](/doc/command-reference/run#for-displaying-and-comparing-data-science-experiments)
+of `dvc run` for more details.
+
+### Iterating over params with foreach stages
+
+Iterating over params was a frequently requested feature. Now users can define
+multiple similar stages with a templatized command.
+
+```yaml
+stages:
+  build:
+    foreach:
+      gb:
+        thresh: 15
+        filename: 'model-gb.hdf5'
+      us:
+        thresh: 10
+        filename: 'model-us.hdf5'
+    do:
+      cmd: >-
+        python script.py --out ${item.filename} --thresh ${item.thresh}
+      outs:
+        - ${item.filename}
+```
+
+## Lightweight ML experiments
+
+DVC uses Git as a foundation for ML experiments. This solid foundation makes
+each ML experiment reproducible and accessible from Git history. This Git-based
+approach works very well for ML projects with mature ML models when only a few
+new experiments per day are running. However, in more active development when
+dozens or hundreds of experiments need to be run in a single day, Git creates
+overhead - each experiment run requires additional Git commands
+`git add/commit`, and comparing all experiments is difficult.
+
+We introduce lightweight experiments in DVC 2.0! This is the way of
+auto-tracking without any overhead from ML engineers.
+
+⚠️ Note, ML experiment is an experimental feature in the coming release. It
+means the commands might change a bit even after the release.
+
+Run an ML experiment with a new hyperparameter from `params.yaml`:
```

Comment:
Should we add `` `dvc exp run` `` in this line so it auto-links to the cmd ref. when that is published? Rn there are very few links to the commands.

Reply:
Same for the following paragraphs before each code block. Could mention each command (`apply/branch/gc`).

Reply:
good idea!